frontend/src/pages/UserAddPage.jsx

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router';
+import { useUser } from '../hooks/useUser';
+import { useApi } from '../hooks/useApi';
+import Spinny from '../components/Spinny';
+
+const UserAddPage = () => {
+  const navigate = useNavigate();
+  const { isLogged, isAdmin } = useUser();
+  const [username, setUsername] = useState('');
+  const [name, setName] = useState('');
+  const [password, setPassword] = useState('');
+  const [repeatPassword, setRepeatPassword] = useState('');
+  const [isNewAdmin, setIsNewAdmin] = useState(false);
+  const [message, setMessage] = useState('');
+  const [error, setError] = useState('');
+
+  const { data, isLoading, fetchFromApi } = useApi({ method: 'POST' });
+
+  useEffect(() => {
+    if (!isLogged) {
+      navigate('/login');
+      return;
+    }
+    if (!isAdmin) {
+      navigate('/');
+    }
+  }, [isLogged, isAdmin, navigate]);
+
+  useEffect(() => {
+    if (data) {
+      if (data.success) {
```

Comment:
```suggestion
    if (response && data) {
      if (response.ok) {
```
To be extra sure, the `response` should be checked.